### PR TITLE
[batch] fix batch create validation

### PR DESCRIPTION
**Pull request by cseed**
We'd mixed up nullable and required.  The default for nullable is False.  Added test.

This fixes the 500 Konrad ran into.

### CODE REVIEW

File: batch/test/test_batch.py [hunk 1]
```diff
@@ -460,3 +460,29 @@ def test_create_idempotence(self):
...
+            # n_jobs None/missing
+            {'billing_project': 'foo', 'n_jobs': None, 'token': 'baz'},
+            {'billing_project': 'foo', 'token': 'baz'},
+            # n_jobs wrong type
+            {'billing_project': 'foo', 'n_jobs': '5', 'token': 'baz'},
+            # token None/missing
+            {'billing_project': 'foo', 'n_jobs': 5, 'token': None},
+            {'billing_project': 'foo', 'n_jobs': 5},
+            # callback None
+            {'billing_project': 'foo', 'callback': None, 'n_jobs': 5, 'token': 'baz'},
```
Comment (jigold) [inline batch/test/test_batch.py]: Don't we want to allow callback to be None? What if there's no callback?

Comment (cseed) [reply batch/test/test_batch.py]: I was thinking no callback URL would be represented as no callback field.  I guess we could treat a null callback in the same way.  This is annoying and I don't think we're super consistent about this overall.  I'll fix, permissive on input seems better.

**Comment by jigold** [comment]
```
io/test/test_batch.py::Test::test_batch_create_validation FAILED
______________________ Test.test_batch_create_validation _______________________

self = <test.test_batch.Test testMethod=test_batch_create_validation>

    def test_batch_create_validation(self):
        bad_configs = [
            # unexpected field fleep
            {'billing_project': 'foo', 'n_jobs': 5, 'token': 'baz', 'fleep': 'quam'},
            # billing project None/missing
            {'billing_project': None, 'n_jobs': 5, 'token': 'baz'},
            {'n_jobs': 5, 'token': 'baz'},
            # n_jobs None/missing
            {'billing_project': 'foo', 'n_jobs': None, 'token': 'baz'},
            {'billing_project': 'foo', 'token': 'baz'},
            # n_jobs wrong type
            {'billing_project': 'foo', 'n_jobs': '5', 'token': 'baz'},
            # token None/missing
            {'billing_project': 'foo', 'n_jobs': 5, 'token': None},
            {'billing_project': 'foo', 'n_jobs': 5},
            # attribute key/value None
            {'attributes': {None: 'v'}, 'billing_project': 'foo', 'n_jobs': 5, 'token': 'baz'},
            {'attributes': {'k': None}, 'billing_project': 'foo', 'n_jobs': 5, 'token': 'baz'},
        ]
        url = deploy_config.url('batch', '/api/v1alpha/batches/create')
        headers = service_auth_headers(deploy_config, 'batch')
        for config in bad_configs:
            r = requests.post(url, json=config, allow_redirects=True, headers=headers)
>           assert r.status_code == 400, (config, r)
E           AssertionError: ({'attributes': {None: 'v'}, 'billing_project': 'foo', 'n_jobs': 5, 'token': 'baz'}, <Response [403]>)
E           assert 403 == 400
E             -403
E             +400

io/test/test_batch.py:487: AssertionError
```

**Comment by cseed** [comment]
Yeah, so it looks like nullable=False isn't working in a keyschema.  I'm just going to rewrite this by hand like I did for jobs.

**Comment by cseed** [comment]
OK, I figured out what was happening.  The problem wasn't with cerberus (although I'm happy to with my change), it is that json.dump always converts a dictionary key into a string.  I had with a key None, and it got turned into the string 'null' in json, because json object values must string keys:

```
>>> import json
>>> d = {None: 5, 'foo': None}
>>> json.loads(json.dumps(d))
{'null': 5, 'foo': None}
```

I remove the broken test.

Note, I pushed two more changes that probably need a proper review:
 - moved jobs validation to batch (from batch_client), I'd been meaning to do that,
 - and I wrote the batch validator explicit in the style of the jobs validator (I'd be meaning to do that, too).

**Pull request merged**